src/librustc_typeck/rscope.rs

```diff
@@ -50,6 +51,71 @@ pub trait RegionScope {
     /// computing `object_lifetime_default` (in particular, in legacy
     /// modes, it may not be relevant).
     fn base_object_lifetime_default(&self, span: Span) -> ty::Region;
+
+    /// If this scope allows anonymized types, return the generics in
+    /// scope, that anonymized types will close over. This property is
+    /// controlled by the region scope because it's fine-grained enough
+    /// to allow restriction of anonymized types to the syntactical extent
+    /// of a function's return type.
```

Comment:
@eddyb after I read below I figured out what the comment meant. I think something like this would have helped me:

-----

If this scope allows anonymized types, return the generics in scope that anonymized types will close over. For example, if you have a function like:

```
fn foo<'a, T>() -> impl Trait { ... }
```

then, for the scope that is used when handling the return type, this function would return `'a` and `T`. This property is controlled by the region scope because it's fine-grained enough to allow restriction of anonymized types to the syntactical extentof a function's return type.